packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,542 @@
+
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  @override
+  WindowType get type => WindowType.regular;
+
+  /// Request change for the window content size.
+  ///
+  /// [contentSize] describes the new requested window size. The properties
+  /// of this object are applied independently of each other. For example,
+  /// setting [WindowSizing.preferredSize] does not affect the [WindowSizing.preferredConstraints]
+  /// set previously.
+  ///
+  /// The platform is free to ignore the request.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void updateContentSize(WindowSizing sizing);
+
+  /// Request change for the window title.
+  /// [title] new title of the window.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void setTitle(String title);
+
+  /// Requests that the window be displayed in its current size and position.
+  ///
+  /// If the window is minimized, the window returns to the size  and position
+  /// that it had before that state was applied. The window will also be
+  /// brought to the top of the window stack.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void activate();
+
+  /// Requests the window to be maximized.
+  ///
+  /// This has no effect if the window is currently full screen or minimized,
+  /// but might affect the window size upon restoring it from minimized or
+  /// full screen state.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void setMaximized(bool maximized);
+
+  /// Returns whether window is currently maximized.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  bool isMaximized();
+
+  /// Requests window to be minimized.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void setMinimized(bool minimized);
+
+  /// Returns whether window is currently minimized.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  bool isMinimized();
+
+  /// Request change for the window to enter or exit fullscreen state.
+  ///
+  /// [fullscreen] whether to enter or exit fullscreen state.
+  /// [display] optional [Display] to use for fullscreen mode.
+  /// The [display] might not be honored by the platform
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void setFullscreen(bool fullscreen, {Display? display});
+
+  /// Returns whether window is currently in fullscreen mode.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  bool isFullscreen();
+}
+
+/// [WindowingOwner] is responsible for creating and managing window controllers.
+///
+/// Custom subclass can be provided by subclassing [WidgetsBinding] and
+/// overriding the [createWindowingOwner] method.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+@internal
+abstract class WindowingOwner {
+  /// Creates a [RegularWindowController] with the provided properties.
+  ///
```

Comment:
App developers might see this and wonder whether they should use this or instantiate `RegularWindowController` directly. I would add a quick explanation that  this used


```suggestion
  /// Most app developers should use [RegularWindowController]'s constructor
  /// instead of calling this method directly. This method allows platforms
  /// to inject platform-specific logic. 
```